jwst/master_background/master_background_step.py

```diff
@@ -0,0 +1,86 @@
+from ..stpipe import Step
+from .. import datamodels
+
+
+__all__ = ["MasterBackgroundStep"]
+
+
+class MasterBackgroundStep(Step):
+    """
+    MasterBackgroundStep:  Compute and subtract master background from spectra
+    """
+
+    spec = """
+        user_background = str(default=None) # Path to user-supplied master background
+        save_background = boolean(default=False) # Save computed master background
+    """
+
+
+    def process(self, input):
```

Comment:
Should `user_background` and `save_background` be arguments to `process`?

Reply:
They are not for any of our other steps.  I was also looking at this.  The arguments get passed via `call` or `run` whose args are generated on-the-fly from the `configobj` parsing of the config file or `spec`.  I hate this, as it is very non-standard python, but it is currently what we do.

There's probably nothing wrong with allowing calls to process, but all the args to `call` are class attributes, so they are accessible within `process` via `self.arg`.

Reply:
Btw, I also dislike that `process` is a public method.  Since it doesn't take args, and we don't generally want users to use it instead of `call` or `run`, it probably should not be public.  `_process` would be better.

It also annoys me that steps do not have an `__init__` method.  Many of our steps set up class attributes within `process` that are not setup via the `Step.__init__` superclass.  An `__init__` method would be natural location to put these, especially if there are other methods other than `process` within the step were these attributes are needed.